### DISCUSSION TITLE
How to fill ID automatically in budibase , now i have to write id manually ! ?

### DISCUSSION THREAD

**Discussion by musaabMD** [Budibase/budibase#12433] [Help] [answered]
Hi
How to automate ID filling automatically in budibase , now i have to write id manually ! ? 

![Screenshot 2023-11-23 at 9 48 34 PM](https://github.com/Budibase/budibase/assets/10286835/5251d078-5ab3-4a00-98f9-1a7070e03328)

**Comment by ConorWebb96** [comment, marked as answer]
Hey @musaabMD,

What type of database are you using? If you are using an internal Budibase database there is a column option called autoID. This will increment the id by 1 with each new record on record creation. I've supplied some images showing how to add this.

<img width="376" alt="Screenshot 2023-11-24 at 07 43 33" src="https://github.com/Budibase/budibase/assets/126772285/c3908202-1fc0-4003-b804-0034e5411647">
<img width="353" alt="Screenshot 2023-11-24 at 07 43 39" src="https://github.com/Budibase/budibase/assets/126772285/512c50cd-fb38-42b6-8e42-ee4bd1fc7b71">

I hope this helps!
